src/GafferImageTestModule/GafferImageTestModule.cpp

```diff
@@ -118,6 +120,36 @@ void testEditableScopeForFormat()
 	);
 }
 
+void validateVisitPixels( GafferImage::Sampler &sampler, const Imath::Box2i &region )
+{
+	int i = 0;
+	int sizeX = region.size().x;
+	sampler.visitPixels( region,
+		[&region, &sampler, &i, sizeX ] ( float value, int x, int y )
+		{
+			int expectedX = ( i % sizeX ) + region.min.x;
+			int expectedY = ( i / sizeX ) + region.min.y;
+			if( x != expectedX || y != expectedY )
+			{
+				throw IECore::Exception( fmt::format (
+					"visitPixels passed incorrect coordinate - expected {},{}, received {},{}",
+					expectedX, expectedY, x, y
+				) );
+			}
+
+			float expectedValue = sampler.sample( x, y );
+			if( value != expectedValue )
+			{
+				throw IECore::Exception( fmt::format(
+					"visitPixels passed incorrect value for pixel {},{} - expected {} received {}",
+					x, y, expectedValue, value
+				) );
+			}
+			i++;
+		}
+	);
```

Comment:
The `expectedX, expectedY` check doesn't guarantee that we actually visited all the pixels, or that we didn't visit additional pixels beyond the region in Y. If we did a final `i == areaOfRegion` check here then we'd be covering those bases.